File: server/internal/model/character.go
// server/internal/model/character.go
package model

import "time"

// Character 角色模型表，用于人物一致性管理
// ReferenceImages 和 Attributes 使用 JSON 字符串存储，保持灵活性
type Character struct {
	ID              uint      `gorm:"primaryKey" json:"id"`
	PortfolioID     uint      `gorm:"index;not null" json:"portfolio_id"`
	Name            string    `gorm:"size:100;not null" json:"name"`
	Description     string    `gorm:"type:text" json:"description"`
	ReferenceImages string    `gorm:"type:json" json:"reference_images"` // JSON 数组：参考图路径列表
	LoraPath        string    `gorm:"size:500" json:"lora_path"`         // LoRA 模型路径（预留）
	Attributes      string    `gorm:"type:json" json:"attributes"`       // JSON 对象：角色属性（发型、服装等）
	CreatedAt       time.Time `json:"created_at"`
	UpdatedAt       time.Time `json:"updated_at"`
}

File: web/src/api/character.ts
// web/src/api/character.ts
import request from './request'

export interface Character {
  id: number
  portfolio_id: number
  name: string
  description: string
  reference_images: string
  attributes: string
  created_at: string
  updated_at: string
}

export interface CreateCharacterPayload {
  name: string
  description?: string
  attributes?: Record<string, string>
}

export const characterApi = {
  list: (portfolioId: number) =>
    request.get(`/portfolios/${portfolioId}/characters`),
  get: (id: number) => request.get(`/characters/${id}`),
  create: (portfolioId: number, data: CreateCharacterPayload) =>
    request.post(`/portfolios/${portfolioId}/characters`, data),
  update: (id: number, data: Partial<CreateCharacterPayload>) =>
    request.put(`/characters/${id}`, data),
  delete: (id: number) => request.delete(`/characters/${id}`),
  uploadReference: (id: number, file: File) => {
    const formData = new FormData()
    formData.append('file', file)
    return request.post(`/characters/${id}/reference`, formData, {
      headers: { 'Content-Type': 'multipart/form-data' },
    })
  },
}
